tests/unit/workflow/test_data_object.py

```diff
@@ -61,3 +63,23 @@ class DataclassesTester(DataObject):
     tester = DataclassesTester(z=False, a="foobar", b=0.3)
     serialized = tester._to_dict()
     assert list(serialized.keys()) == ["b", "a", "z"]
+
+
+def test__data_object__extras() -> None:
+    class Config:
+        extra = Extra.allow
+
+    @dataclass(frozen=True, config=Config)
```

Comment:
@gordonhart What do you think of using `pydantic.dataclass` with `extra` (depends on 1.10 above) .

Reply:
This looks like it could be exactly what we want. Are there additional tests we can write to see how well this works for different cases?

Reply:
Do you know what in particular is not supported about `extra="allow"`? From [Pydantic docs](https://docs.pydantic.dev/latest/usage/dataclasses/#dataclass-config):

<img width="791" alt="Screen Shot 2023-09-06 at 9 34 21 PM" src="https://github.com/kolenaIO/kolena/assets/9648886/178372c0-932c-4b5e-9f38-bdc386826bd6">